<launch>
    # --------------------------------------------------------------------- PARAMETER --------------------------------------------------------------------- 
    
    ## General Parameters
    <env name="GAZEBO_MODEL_PATH" value="${GAZEBO_MODEL_PATH}:$(find rotors_gazebo)/models:$(find idmind_arco)/models"/>
    <env name="GAZEBO_RESOURCE_PATH" value="${GAZEBO_RESOURCE_PATH}:$(find rotors_gazebo)/models:$(find idmind_arco)/models"/>

    <param name="use_sim_time" value="true" />

    ## UAV Parameter
    <arg name="enable_logging" default="false"/>
    <arg name="enable_ground_truth" default="false"/>
    <arg name="mav_name" default="firefly" />
    <arg name="uav_name" default="dji_210" />
    <arg name="ugv_name" default="arco" />
    <arg name="ugv_x" default="-16.0" />
    <arg name="ugv_y" default="-3.5" />
    <arg name="ugv_z" default="0.1" />
    <arg name="ugv_th" default="0.5" />
    <arg name="uav_z" default="0.43" />

    ## UGV Parameter Define real/simulation and controller type
    <arg name="controller" value="controller" />                              <!-- Controller - joystick, move_base or controller -->
    <arg name="config" value="lispolis" />    
    <arg name="simulation" value="true" />                                  <!-- Define if it will run on a robot or simulation -->
    
    # --------------------------------------------------------------------- NODES --------------------------------------------------------------------- 
    
    # ---------------------------------------------------Start Gazebo environment ---------------------------------------------------
    <include file="$(find idmind_arco)/launch/empty_world.launch">
        <arg name="world_name" default="$(find idmind_arco)/worlds/$(arg config)_simul.world" />
        <arg name="paused" value="true" />
        <arg name="gui" value="true" />
        <arg name="debug" value="false" />
        <arg name="verbose" value="true" />
    </include>

    <node pkg="tf" type="static_transform_publisher" name="tf_world" args="0.0 0.0 -0.0 0.0 -0.0 0.0  world  map 10"/> 

    # --------------------------------------------------- Start UGV Simulation ---------------------------------------------------
    ## UGV Gazebo
    <include file="$(find idmind_arco)/launch/start_robot.launch">
        <arg name="controller" value="$(arg controller)" />
        <arg name="simulation" value="true" />
        <arg name="config" value="$(arg config)" />
        <!-- <arg name="robot_name" value="$(arg ugv_name)" /> -->
        <arg name="robot_name" value="firefly" />
        <arg name="mapping" value="false" />
        <arg name="init_x" value="$(arg ugv_x)" />
        <arg name="init_y" value="$(arg ugv_y)" />
        <arg name="init_z" value="$(arg ugv_z)" />
        <arg name="init_th" value="$(arg ugv_th)" />
        <arg name="use_tether" value="true" />
    </include>
    
    ## UGV Localization Stuff
    <group ns="$(arg ugv_name)">
        <node pkg="timed_roslaunch" type="timed_roslaunch.sh" name="timed_ugv_dll" output="screen"
            args="10 marsupial_optimizer dll.launch robot_name:=$(arg ugv_name)  initial_x:=12.0 initial_y:=20.08 initial_z:=0.3 initial_a:=0.0">
        </node>
    </group>

        
        
        
</launch>